config/migrate.ts

```diff
@@ -1,218 +1,119 @@
-    writeConfig({ source: updatedConfigJson });
-    config_DEPRECATED.deleteConfigFile();
-  } catch (error) {
-    deleteEmptyConfigFile();
-    throw new Error(
-      i18n(`${i18nKey}.errors.writeConfig`, { configPath: GLOBAL_CONFIG_PATH }),
-      { cause: error }
-    );
-  }
+  return parseConfig(configFileSource);
 }
 
-export function migrateConfig(
-  deprecatedConfig: CLIConfig_DEPRECATED | null
-): void {
-  if (!deprecatedConfig) {
-    throw new Error(i18n(`${i18nKey}.errors.noDeprecatedConfig`));
-  }
-  const { defaultPortal, portals, ...rest } = deprecatedConfig;
-  const updatedConfig = {
-    ...rest,
-    defaultAccount: defaultPortal,
-    accounts: portals
-      .filter(({ portalId }) => portalId !== undefined)
-      .map(({ portalId, ...rest }) => ({
-        ...rest,
-        accountId: portalId!,
-      })),
-  };
-  writeGlobalConfigFile(updatedConfig, true);
+export function migrateConfigAtPath(path: string): void {
+  createEmptyConfigFile(true);
+  const configToMigrate = getConfigAtPath(path);
+  writeConfigFile(configToMigrate, getGlobalConfigFilePath());
+  fs.unlinkSync(path);
 }
 
-type ConflictValue = boolean | string | number | CmsPublishMode | Environment;
 export type ConflictProperty = {
-  property: keyof CLIConfig_NEW;
-  oldValue: ConflictValue;
-  newValue: ConflictValue;
+  property: keyof HubSpotConfig;
+  oldValue: ValueOf<HubSpotConfig>;
+  newValue: ValueOf<HubSpotConfig>;
 };
 
 export function mergeConfigProperties(
-  globalConfig: CLIConfig_NEW,
-  deprecatedConfig: CLIConfig_DEPRECATED,
+  toConfig: HubSpotConfig,
+  fromConfig: HubSpotConfig,
   force?: boolean
 ): {
-  initialConfig: CLIConfig_NEW;
+  configWithMergedProperties: HubSpotConfig;
   conflicts: Array<ConflictProperty>;
 } {
-  const propertiesToCheck: Array<keyof Partial<CLIConfig>> = [
-    DEFAULT_CMS_PUBLISH_MODE,
-    HTTP_TIMEOUT,
-    ENV,
-    HTTP_USE_LOCALHOST,
-    ALLOW_USAGE_TRACKING,
-  ];
   const conflicts: Array<ConflictProperty> = [];
 
-  propertiesToCheck.forEach(prop => {
-    if (prop in globalConfig && prop in deprecatedConfig) {
-      if (force || globalConfig[prop] === deprecatedConfig[prop]) {
-        // @ts-expect-error Cannot reconcile CLIConfig_NEW and CLIConfig_DEPRECATED types
-        globalConfig[prop] = deprecatedConfig[prop];
-      } else {
+  if (force) {
+    toConfig.defaultCmsPublishMode = fromConfig.defaultCmsPublishMode;
+    toConfig.httpTimeout = fromConfig.httpTimeout;
+    toConfig.env = fromConfig.env;
+    toConfig.httpUseLocalhost = fromConfig.httpUseLocalhost;
+    toConfig.allowUsageTracking = fromConfig.allowUsageTracking;
+    toConfig.defaultAccount = fromConfig.defaultAccount;
+  } else {
+    const propertiesToCheck = [
+      DEFAULT_CMS_PUBLISH_MODE,
+      HTTP_TIMEOUT,
+      ENV,
+      HTTP_USE_LOCALHOST,
+      ALLOW_USAGE_TRACKING,
+      DEFAULT_ACCOUNT,
+    ] as const;
+
+    propertiesToCheck.forEach(prop => {
+      if (toConfig[prop] !== fromConfig[prop]) {
```

Comment:
The from config value could be undefined here (I think). Should we only count it as a conflict if it has a value?

Reply:
Oh yes good call